tools/kraken.wdl

```diff
@@ -0,0 +1,300 @@
+        String db_name = "kraken2_"+library_name+"_library"
+        Boolean clean_up = true
+        Int memory_gb = 4
+        Int added_disk_size_gb = 0
+        Int max_retries = 3
+    }
+
+    Int disk_size_gb = (if library_name=="bacteria" then 300 else 20) + added_disk_size_gb
+
+    command <<<
+        set -euo pipefail
+
+        kraken2-build --download-library \
+            ~{library_name} \
+            --use-ftp \
+            --db ~{db_name} 2>&1 \
+            | awk '/gunzip:/ { print; exit 42 } !/gunzip:/ { print }' 1>&2
+
+        tar -C ~{db_name}/ -czf "~{db_name}.tar.gz" .
+
+        if [ "~{clean_up}" = "true" ]; then
+            rm -r ~{db_name}
+        fi
+    >>>
+ 
+    runtime {
+        memory: memory_gb + " GB"
+        disk: disk_size_gb + " GB"
+        cpu: 1
+        docker: 'ghcr.io/stjudecloud/kraken2:branch-kraken2-1.0.0'
+        maxRetries: max_retries
+    }
+
+    output {
+        File library = db_name + ".tar.gz"
+    }
+
+    meta {
+        author: "Andrew Frantz"
+        email: "andrew.frantz@stjude.org"
+        description: "This WDL task downloads a Kraken2 library."
+    }
+}
+
+task create_library_from_fastas {
+    input {
+        Array[File] fastas
+        String db_name = "kraken2_custom_library"
+        Boolean clean_up = true
+        Int memory_gb = 4
+        Int added_disk_size_gb = 0
+        Int max_retries = 1
+    }
+
+    Int fastas_size = ceil(size(fastas, "GiB"))
+    Int disk_size_gb = fastas_size * 5 + added_disk_size_gb
+
+    command <<<
+        set -euo pipefail
+
+        >&2 echo "*** start adding custom FASTAs ***"
+        echo "~{sep="\n" fastas}" > fastas.txt
+        while read -r fasta; do
+            gunzip -c "$fasta" > tmp.fa
+            kraken2-build --add-to-library tmp.fa --db ~{db_name}
+        done < fastas.txt
+        rm tmp.fa
+        >&2 echo "*** done adding custom FASTAs ***"
+
+        tar -C ~{db_name}/ -czf "~{db_name}.tar.gz" .
+
+        if [ "~{clean_up}" = "true" ]; then
+            rm -r ~{db_name}
+        fi
+    >>>
+ 
+    runtime {
+        memory: memory_gb + " GB"
+        disk: disk_size_gb + " GB"
+        cpu: 1
+        docker: 'quay.io/biocontainers/kraken2:2.1.2--pl5321h9f5acd7_2'
+        maxRetries: max_retries
+    }
+
+    output {
+        File custom_library = db_name + ".tar.gz"
+    }
+
+    meta {
+        author: "Andrew Frantz"
+        email: "andrew.frantz@stjude.org"
+        description: "This WDL task adds custom FASTAs to a Kraken2 DB."
+    }
+}
+
+task build_db {
+    input {
+        Array[File] tarballs
+        String db_name = "kraken2_db"
+        Int kmer_len = 35
+        Int minimizer_len = 31
+        Int minimizer_spaces = 7
+        Int max_db_size_gb = -1
+        Float load_factor = 0.7
+        Boolean clean_up = true
+        Int added_memory_gb = 0
+        Int added_disk_size_gb = 0
+        Int ncpu = 1
+        Boolean detect_nproc = false
+        Int max_retries = 1
+    }
```

Comment:
Will do, good call. Although I don't think we need the `clean_up` option anymore. I added it in case I needed to debug something, but I never wound up looking at those files. I think it should always be safe to clean out the files after the tar is complete.